### ISSUE TITLE
fetch_repodata_remote_request raises RepodataOnDisk exception

### ISSUE THREAD

**Issue by mcg1969** [conda/conda#13595]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

The `fetch_repodata_remote_request` was deprecated in 23.9, and is scheduled for removal in conda 24.3. Unfortunately, it it seems that it is now broken:

```
>>> from conda.core.subdir_data import fetch_repodata_remote_request
>>> fetch_repodata_remote_request('https://repo.anaconda.com/pkgs/main/linux-64', None, None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mgrant/miniconda3/lib/python3.11/site-packages/conda/deprecations.py", line 71, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mgrant/miniconda3/lib/python3.11/site-packages/conda/core/subdir_data.py", line 560, in fetch_repodata_remote_request
    raw_repodata_str = subdir._repo.repodata(cache_state)  # type: ignore
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mgrant/miniconda3/lib/python3.11/site-packages/conda/gateways/repodata/jlap/interface.py", line 54, in repodata
    self.repodata_parsed(state)
  File "/Users/mgrant/miniconda3/lib/python3.11/site-packages/conda/gateways/repodata/jlap/interface.py", line 121, in repodata_parsed
    raise RepodataOnDisk()
conda.gateways.repodata.RepodataOnDisk
```

### Conda Info

```shell
active environment : base
    active env location : /Users/mgrant/miniconda3
            shell level : 1
       user config file : /Users/mgrant/.condarc
 populated config files : /Users/mgrant/.condarc
          conda version : 24.1.1
    conda-build version : 3.28.3
         python version : 3.11.7.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=m1
                          __conda=24.1.1=0
                          __osx=14.3=0
                          __unix=0=0
       base environment : /Users/mgrant/miniconda3  (writable)
      conda av data dir : /Users/mgrant/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-arm64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-arm64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/mgrant/miniconda3/pkgs
                          /Users/mgrant/.conda/pkgs
       envs directories : /Users/mgrant/miniconda3/envs
                          /Users/mgrant/.conda/envs
               platform : osx-arm64
             user-agent : conda/24.1.1 requests/2.31.0 CPython/3.11.7 Darwin/23.3.0 OSX/14.3 solver/libmamba conda-libmamba-solver/23.12.0 libmambapy/1.5.6 aau/0.4.3
                UID:GID : 502:20
             netrc file : None
           offline mode : False
```

### Conda Config

```shell
==> /Users/mgrant/.condarc <==
anaconda_anon_usage: True
```

**Comment by jezdez** [comment]
Thanks for the report @mcg1969! This seems to be a regression introduced while JLAP has been integrated into the repodata interfaces.

@dholth Seems like `repodata` isn't returning the cached repodata string at all? Let's prepare this for a patch release.

**Comment by dholth** [comment]
This should also be avoidable by setting `--no-repodata-use-zst`. We can override that setting as part of a patched `fetch_repodata_remote_request`.

**Comment by mcg1969** [comment]
While working on the patch, I'd appreciate if we can clarify the deprecation commentary better to describe more fully how to replace `fetch_repodata_remote_request`. The comments point to the class `conda.core.subdir_data.SubdirData`, but provide no further guidance on what methods to call. Then again, if the fixed `fetch_repodata_remote_request` is cleanly implemented that will be its own documentation!

**Comment by dholth** [comment]
```
subdir = SubdirData(Channel(url), repodata_fn=repodata_fn)
fetcher = subdir.repo_fetch
repodata, state = fetcher.fetch_latest_parsed()
```

**Comment by kenodegard** [comment]
Fixed in #13596, to be included in `conda 24.1.2` patch release
